pkg/vault/test/foundry/E2eSwap.t.sol

```diff
@@ -0,0 +1,516 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.24;
+
+import "forge-std/Test.sol";
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import { IAuthentication } from "@balancer-labs/v3-interfaces/contracts/solidity-utils/helpers/IAuthentication.sol";
+import { IProtocolFeeController } from "@balancer-labs/v3-interfaces/contracts/vault/IProtocolFeeController.sol";
+
+import { ArrayHelpers } from "@balancer-labs/v3-solidity-utils/contracts/helpers/ArrayHelpers.sol";
+import { ERC20TestToken } from "@balancer-labs/v3-solidity-utils/contracts/test/ERC20TestToken.sol";
+import { FixedPoint } from "@balancer-labs/v3-solidity-utils/contracts/math/FixedPoint.sol";
+
+import { PoolMock } from "../../contracts/test/PoolMock.sol";
+import { BaseVaultTest } from "./utils/BaseVaultTest.sol";
+
+contract E2eSwapTest is BaseVaultTest {
+    using ArrayHelpers for *;
+
+    ERC20TestToken internal tokenA;
+    ERC20TestToken internal tokenB;
+    uint256 internal tokenAIdx;
+    uint256 internal tokenBIdx;
+    address internal sender;
+    address internal poolCreator;
+
+    uint256 internal minPoolSwapFeePercentage;
+    uint256 internal maxPoolSwapFeePercentage;
+
+    uint256 internal minSwapAmountTokenA;
+    uint256 internal maxSwapAmountTokenA;
+
+    uint256 internal minSwapAmountTokenB;
+    uint256 internal maxSwapAmountTokenB;
+
+    function setUp() public virtual override {
+        BaseVaultTest.setUp();
+
+        IProtocolFeeController feeController = vault.getProtocolFeeController();
+        IAuthentication feeControllerAuth = IAuthentication(address(feeController));
+
+        authorizer.grantRole(
+            feeControllerAuth.getActionId(IProtocolFeeController.setGlobalProtocolSwapFeePercentage.selector),
+            admin
+        );
+
+        _setUpVariables();
+
+        // Set protocol and creator fees to 50%, so we can measure the charged fees.
+        vm.prank(admin);
+        feeController.setGlobalProtocolSwapFeePercentage(50e16);
+
+        vm.prank(poolCreator);
+        // Set pool creator fee to 100%, so protocol + creator fees equals the total charged fees.
+        feeController.setPoolCreatorSwapFeePercentage(pool, FixedPoint.ONE);
+
+        (tokenAIdx, tokenBIdx) = _getTokenIndexes();
+
+        // Donate tokens to vault, so liquidity tests are possible.
+        tokenA.mint(address(vault), 100 * poolInitAmount);
```

Comment:
At `100 * poolInitAmount` the same

Reply:
I think, since the test is prepared to deal with different decimals now, creating a constant for 100 * poolInitAmount doesn't make much sense anymore (because poolInitAmount is different, depending on the decimals)